.github/workflows/e2e_android.new.yml

```diff
@@ -0,0 +1,189 @@
+
+    - name: Install JS dependencies
+      run: |
+        npm install
+        # rm -rf node_modules/react-native-pure-jwt
+        # npm install react-native-pure-jwt
+        # git status
+
+    - name: Download the small example cv and geomodel
+      run: |
+        npm run add-example-model -- -f=main
+
+    - name: Android Build
+      run: |
+        cd android
+        ./gradlew build -PreactNativeArchitectures=x86 -x lint -x lintVitalRelease 
+
+    - name: Upload APK
+      uses: actions/upload-artifact@v5
+      with:
+        name: release-apk
+        path: android/app/build/outputs/apk/release/*-release.apk
+    
+  test:
+    runs-on: ubuntu-24.04-m
+    needs: build
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Download APK
+        uses: actions/download-artifact@v6
+        with:
+          name: release-apk
+          path: e2e-build
+
+      - name: Enable KVM
+        run: |
+          echo 'KERNEL=="kvm", GROUP="kvm", MODE="0666", OPTIONS+="static_node=kvm"' | sudo tee /etc/udev/rules.d/99-kvm4all.rules
+          sudo udevadm control --reload-rules
+          sudo udevadm trigger --name-match=kvm
+
+      - name: Install Maestro
+        run: |
+          export MAESTRO_VERSION="2.0.8"; curl -Ls "https://get.maestro.mobile.dev" | bash
+          echo "$HOME/.maestro/bin" >> "$GITHUB_PATH"
+
+      - name: Attempt to restore AVD cache
+        uses: actions/cache/restore@v4
+        id: avd-cache-restore
+        with:
+          path: |
+            ~/.android/avd/*
+            ~/.android/adb*
+          key: avd-29
+
+      - name: Create AVD and generate snapshot for caching
+        if: steps.avd-cache-restore.outputs.cache-hit != 'true'
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          cores: 4
+          api-level: 29
+          target: google_apis
+          force-avd-creation: false
+          emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+          disable-animations: false
+          script: echo "Generated AVD snapshot for caching."
+
+      - name: Save AVD cache
+        if: steps.avd-cache-restore.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v4
+        with:
+          path: |
+            ~/.android/avd/*
+            ~/.android/adb*
+          key: avd-29
+
+      - name: run tests
+        uses: reactivecircus/android-emulator-runner@v2
+        env:
+          MAESTRO_DRIVER_STARTUP_TIMEOUT: 60000
+          MAESTRO_CLI_NO_ANALYTICS: 1
+          MAESTRO_RECORD: true
+        with:
+          cores: 4
+          api-level: 29
+          target: google_apis
+          force-avd-creation: false
+          emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+          disable-animations: true
+          script: |
+            adb install -r e2e-build/*.apk
+            maestro test --no-ansi e2e/maestro/android/signedOut.yaml
```

Comment:
This is our first and only "real" use of [reactivecircus/android-emulator-runner](https://github.com/ReactiveCircus/android-emulator-runner/tree/v2) which orchestrates an emulator available to our `script` step that installs our apk to the emulator and runs our single maestro test against it.

This action (along with it's [apparently associated GH-paired kvm affordance ](https://github.blog/changelog/2023-02-23-hardware-accelerated-android-virtualization-on-actions-windows-and-linux-larger-hosted-runners/)) made a _huge_ difference in even making the emulator _possible_ to run on the 2/4 core ubuntu runners, let alone execute tests.